src/AppStateContext.tsx

```diff
@@ -290,15 +288,34 @@ export const AppStateProvider = ({ children }: React.PropsWithChildren<{}>) => {
     //     }
     // });
 
-      //Lets' subscribe on the default channel.
-      state.pubnub.subscribe(
-        {
-          channels: [state.channel], //Only one channel, split in different rows if required and load in props, can be set by load balancer.
-          withPresence: state.presence
-        }
+      //Get the history on the default channel.
+      state.pubnub.history(
+          {
+              channel: state.channel,
+              count: 100
+          },
+          (status, response) => {
+            if (typeof response.messages !== "undefined" && response.messages.length > 0) {
+              for (var i = 0; i <= response.messages.length; i++) {
```

Comment:
This is really inefficient - this will dispatch and update the state up to 100 times in a rapid succession. I think passing in all messages in one dispatch call would be way safer here

Reply:
@nfodorpubnub Can we modify the ADD_HISTORY call to accept an array of messages from history instead of individual messages?